programs/openbook-v2/tests/program_test/mod.rs

```diff
@@ -248,4 +274,121 @@ impl TestContext {
     pub async fn new() -> Self {
         TestContextBuilder::new().start_default().await
     }
+
+    #[allow(clippy::too_many_arguments)]
+    pub async fn new_with_market(
```

Comment:
lgtm, but given that (almost) the same set of parameters is always used, you can further reduce boilerplate by using a new struct as input argument with a custom `default` impl.